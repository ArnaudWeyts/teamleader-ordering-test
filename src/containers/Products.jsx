import React, { Component } from 'react';
import { connect } from 'react-redux';

import { fetchProducts } from '../actions/productActions';
import Product from '../components/Product';

class Products extends Component {
  componentDidMount() {
    const { dispatch } = this.props;
    dispatch(fetchProducts());
  }

  render() {
    const { isFetching, products, addToOrder } = this.props;

    if (isFetching) {
      return <div>Loading...</div>;
    }

    return (
      <div>
        {products.map(product => (
          <div key={product.id}>
            <Product product={product} />
            <button type="button" onClick={() => addToOrder(product.id)}>
              Add to order
            </button>
          </div>
        ))}
      </div>
    );
  }
}

const mapStateToProps = state => ({
  isFetching: state.products.isFetching,
  products: state.products.data
});

export default connect(mapStateToProps)(Products);
