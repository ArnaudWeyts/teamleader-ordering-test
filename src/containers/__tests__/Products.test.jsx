import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import Products from '../Products';

const mockStore = configureStore([thunk]);

it('renders without crashing', () => {
  const store = mockStore({ products: { data: [] } });
  shallow(<Products store={store} />);
});

it('renders the products loading', () => {
  const store = mockStore({
    products: { data: [] }
  });

  const component = renderer.create(<Products store={store} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders the products', () => {
  const store = mockStore({
    products: {
      data: [
        {
          id: 'A101',
          description: 'Screwdriver',
          category: '1',
          price: 9.75
        },
        {
          id: 'A102',
          description: 'Electric screwdriver',
          category: '1',
          price: 49.5
        }
      ]
    }
  });

  const component = renderer.create(<Products store={store} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
