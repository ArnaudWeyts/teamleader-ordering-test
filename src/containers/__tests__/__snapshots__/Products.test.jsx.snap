// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the products 1`] = `
<div>
  <div>
    <div>
      <p>
        Description: 
        Screwdriver
      </p>
      <p>
        Category: 
        1
      </p>
      <p>
        Price: 
        €9.75
      </p>
    </div>
    <button
      onClick={[Function]}
      type="button"
    >
      Add to order
    </button>
  </div>
  <div>
    <div>
      <p>
        Description: 
        Electric screwdriver
      </p>
      <p>
        Category: 
        1
      </p>
      <p>
        Price: 
        €49.50
      </p>
    </div>
    <button
      onClick={[Function]}
      type="button"
    >
      Add to order
    </button>
  </div>
</div>
`;

exports[`renders the products loading 1`] = `<div />`;
